lib/db/schema/patch-022-023.sql

```diff
@@ -0,0 +1,263 @@
+    uaOS,
+    uaOSVersion,
+    uaDeviceType,
+    createdAt
+  );
+
+  IF tokenVerified = FALSE THEN
+    INSERT INTO unverifiedTokens(
+      tokenVerificationId,
+      tokenId
+    )
+    VALUES(
+      tokenVerificationId,
+      tokenId
+    );
+  END IF;
+
+  COMMIT;
+END;
+
+-- Update createKeyFetchToken stored procedure to
+-- insert into unverifiedTokens.
+CREATE PROCEDURE `createKeyFetchToken_2` (
+  IN tokenId BINARY(32),
+  IN authKey BINARY(32),
+  IN uid BINARY(16),
+  IN keyBundle BINARY(96),
+  IN createdAt BIGINT UNSIGNED,
+  IN tokenVerified BOOLEAN,
+  IN tokenVerificationId BINARY(32)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  INSERT INTO keyFetchTokens(
+    tokenId,
+    authKey,
+    uid,
+    keyBundle,
+    createdAt
+  )
+  VALUES(
+    tokenId,
+    authKey,
+    uid,
+    keyBundle,
+    createdAt
+  );
+
+  IF tokenVerified = FALSE THEN
+    INSERT INTO unverifiedTokens(
+      tokenVerificationId,
+      tokenId
+    )
+    VALUES(
+      tokenVerificationId,
+      tokenId
+    );
+  END IF;
+
+  COMMIT;
+END;
+
+-- Update deleteSessionToken stored procedure to
+-- conditionally delete from unverifiedTokens.
+CREATE PROCEDURE `deleteSessionToken_2` (
+  IN tokenIdArg BINARY(32)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  DELETE FROM sessionTokens WHERE tokenId = tokenIdArg;
+  DELETE FROM unverifiedTokens WHERE tokenId = tokenIdArg;
+
+  COMMIT;
+END;
+
+-- Update deleteKeyFetchToken stored procedure to
+-- conditionally delete from unverifiedTokens.
+CREATE PROCEDURE `deleteKeyFetchToken_2` (
+  IN tokenIdArg BINARY(32)
+)
+BEGIN
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  DELETE FROM keyFetchTokens WHERE tokenId = tokenIdArg;
+  DELETE FROM unverifiedTokens WHERE tokenId = tokenIdArg;
+
+  COMMIT;
+END;
+
+-- Add stored procedure for fetching sessionToken
+-- with its verified state.
+CREATE PROCEDURE `sessionTokenVerified_1` (
+  IN tokenIdArg BINARY(32)
+)
+BEGIN
+  DECLARE tokenVerified BOOLEAN;
+
+  DECLARE EXIT HANDLER FOR SQLEXCEPTION
+  BEGIN
+    ROLLBACK;
+    RESIGNAL;
+  END;
+
+  START TRANSACTION;
+
+  SET tokenVerified = (
+    CASE WHEN EXISTS(
+      SELECT * FROM unverifiedTokens
+      WHERE unverifiedTokens.tokenId = tokenIdArg
+    )
+      THEN FALSE
+      ELSE TRUE
+    END
+  );
+
+  SELECT
+    t.tokenData,
+    t.uid,
+    t.createdAt,
+    t.uaBrowser,
+    t.uaBrowserVersion,
+    t.uaOS,
+    t.uaOSVersion,
+    t.uaDeviceType,
+    t.lastAccessTime,
+    a.emailVerified,
+    a.email,
+    a.emailCode,
+    a.verifierSetAt,
+    a.locale,
+    a.createdAt AS accountCreatedAt,
+    tokenVerified AS tokenVerified
+  FROM sessionTokens AS t
+  INNER JOIN accounts AS a
+  ON t.uid = a.uid
+  WHERE t.tokenId = tokenIdArg;
+
+  COMMIT;
+END;
```

Comment:
In the interests of being explicit: the above approach using an `unverifiedTokens` table, and inserting records for all new tokens by default, seems OK to me on balance.  As long as the API makes it hard to create tokens that are 'verified' when we don't mean them to be.  Thanks @dannycoates!